extensions/amp-iframe/1.0/component.js

```diff
@@ -89,22 +90,28 @@ export function Iframe({
     if (!win) {
       return;
     }
-    const io = new win.IntersectionObserver((entries) => {
-      const last = entries[entries.length - 1];
-      isIntersectingRef.current = last.isIntersecting;
-      if (last.isIntersecting || !dataRef.current || !win) {
-        return;
-      }
-      attemptResize();
-    });
-    io.observe(iframe);
+
     win.addEventListener('message', handlePostMessage);
 
     return () => {
-      io.unobserve(iframe);
       win.removeEventListener('message', handlePostMessage);
     };
-  }, [attemptResize, handlePostMessage]);
+  }, [handlePostMessage]);
+
+  const ioCallback = useCallback(
+    ({isIntersecting}) => {
+      if (isIntersecting === isIntersectingRef.current) {
+        return;
+      }
+      isIntersectingRef.current = isIntersecting;
+      if (!isIntersecting) {
+        attemptResize();
+      }
+    },
+    [attemptResize]
+  );
+
+  const measureRef = useIntersectionObserver(ioCallback);
```

Comment:
This call signature look incorrect?

Reply:
What is wrong about it? I think a ref-returning hook is the only way to pull this off

Reply:
I recently updated `useIntersectionObserver` params to just take a callback, maybe thats why you think it looks incorrect.

Reply:
Ha, I commented on the wrong one. Meant the other one with 3 params.